helix-view/src/editor.rs

```diff
@@ -826,6 +826,9 @@ impl Editor {
                         Some(Action::Close(view.id))
                     }
                 } else {
+                    // documents also need be removed from the view "document access history"
+                    // so we don't accidentally try to jump back to them after they have been deleted
+                    view.docs_access_history.retain(|doc| doc != &doc_id);
```

Comment:
Let's fix this in https://github.com/helix-editor/helix/blob/955a6cd540ca0483859b98512d7404b5f028b8d4/helix-view/src/view.rs#L64-L66=


The method is already called on line 818

Reply:
Ah wait, that's not actually possible since it's a method on `view.jumps`. Perhaps we should add a method to `View` that cleans up both jumps and access history. Any place calling `jumps.remove` should then call this new method instead

Reply:
Ok, I'll have a go at doing it

Reply:
@archseer I've moved the cleanup into a `view.remove_document` function now, take a look when you have a moment.

We could also take this approach further and add an `Editor.remove_document` function which would look like:

```rust
impl Editor {
  fn remove_document(&mut self, doc_id: &DocumentId) {
    for view in self.tree.views_mut() {
      view.remove_document(&doc_id);
    }
    self.documents.remove(&doc_id);
  }
}
```